Makefile

```diff
@@ -201,12 +201,21 @@ out/minikube-linux-aarch64: out/minikube-linux-arm64
 	$(if $(quiet),@echo "  CP       $@")
 	$(Q)cp $< $@
 
-.PHONY: minikube-linux-amd64 minikube-linux-arm64 minikube-darwin-amd64 minikube-windows-amd64.exe
+.PHONY: minikube-linux-amd64 minikube-linux-arm64
 minikube-linux-amd64: out/minikube-linux-amd64 ## Build Minikube for Linux 64bit
-minikube-linux-arm64: out/minikube-linux-arm64 ## Build Minikube for ARM 64bit
-minikube-darwin-amd64: out/minikube-darwin-amd64 ## Build Minikube for Darwin 64bit
+minikube-linux-arm64: out/minikube-linux-arm64 ## Build Minikube for arm 64bit
+
+.PHONY: minikube-darwin minikube-darwin-amd64 minikube-darwin-arm64
+minikube-darwin-amd64: out/minikube-darwin-amd64 ## Build Minikube for Darwin x86 64bit
+minikube-darwin-arm64: out/minikube-darwin-arm64 ## Build Minikube for Darwin arm 64bit
+minikube-darwin: out/minikube-darwin ## Build Minikube fat binary for Darwin (both arm and x86 64bit)
+
+.PHONY: minikube-windows-amd64.exe
 minikube-windows-amd64.exe: out/minikube-windows-amd64.exe ## Build Minikube for Windows 64bit
 
+out/minikube-darwin: out/minikube-darwin-amd64 out/minikube-darwin-arm64
+	lipo -create $^ -output $@
```

Comment:
what is this lipo for ?

Reply:
it's a standard darwin to work with executables.
https://ss64.com/osx/lipo.html
Here it's used to repack binaries for amd64 and arm64 into single executable file

Reply:
does this affect the binary size ? and would it not work without it ?

Reply:
It's a pun on "fat binaries", like https://en.wikipedia.org/wiki/Liposuction
(they later got renamed to "universal binaries", making it incomprehensible)

There is actually a similar project for Linux: https://icculus.org/fatelf/
But it never caught on, I guess not the same amount of closed binaries ?

Reply:
i.e. it makes the "files to download" list shorter, but the binaries bigger (twice)

if we did that for _all_ of kubernetes (all arch), it would be 5x the current size

Reply:
got it. removed